components/match2/wikis/chess/match_summary.lua

```diff
@@ -0,0 +1,114 @@
+---
+-- @Liquipedia
+-- wiki=chess
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local CustomMatchSummary = {}
+
+local Array = require('Module:Array')
+local Lua = require('Module:Lua')
+
+local Eco = Lua.import('Module:Chess/ECO')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Icon = Lua.import('Module:Icon')
+local MatchSummary = Lua.import('Module:MatchSummary/Base')
+local MatchSummaryWidgets = Lua.import('Module:Widget/Match/Summary/All')
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+local KING_ICON_SIZE = '150%'
+local WHITE_KING = Icon.makeIcon{
+	iconName = 'chesskingwhite',
+	additionalClasses = {'show-when-light-mode'},
+	size = KING_ICON_SIZE,
+} .. Icon.makeIcon{
+	iconName = 'chesskingblack',
+	additionalClasses = {'show-when-dark-mode'},
+	size = KING_ICON_SIZE,
+}
+local BLACK_KING = Icon.makeIcon{
+	iconName = 'chesskingblack',
+	additionalClasses = {'show-when-light-mode'},
+	size = KING_ICON_SIZE,
+} .. Icon.makeIcon{
+	iconName = 'chesskingwhite',
+	additionalClasses = {'show-when-dark-mode'},
+	size = KING_ICON_SIZE,
+}
```

Comment:
What? 
Why is the black king white in dark mode? That seems counterproductive

Reply:
Earlier we run into issue where the light/dark mode causes the situation where icon is not visible in one of the modes due to how it flips (I didnt work on this part so) robbie set it up in a way that 

lightmode: Opaque piece = Black, Piece with only outlines = White
darkmode: Reverse of lightmode

i think so

![image](https://github.com/user-attachments/assets/0fac2e8d-2ae1-4fc6-9102-4132fcfcb54f)

Reply:
> What? Why is the black king white in dark mode? That seems counterproductive

because fontawsome full inverts on darkmode, hence black becomes white and vice versa

Reply:
i think this is resolved